ADR-004 Add MkDocs Material Documentation Site

jhagberg opened the pull request:
## Add MkDocs Material Documentation Site

This PR implements [ADR-0004](docs/decisions/0004-use-mkdocs-material-for-documentation-site.md) by setting up a complete MkDocs Material documentation site with SciLifeLab visual identity.

### What's Included

**Documentation Site (`mkdocs.yml`)**
- MkDocs Material theme with SciLifeLab branding (teal/lime colors)
- Open Sans typography matching scilifelab.se
- Light/dark mode toggle
- Mermaid diagram support for architecture visualizations
- Cookie consent for GDPR compliance
- Navigation tabs, search, and code copy features

**Visual Identity (`docs/stylesheets/extra.css`)**
- Custom CSS implementing SciLifeLab color palette (#045C64 teal, #A7C947 lime)
- Styled header, footer, tables, and admonitions
- Consistent branding across light and dark modes

**Homepage (`docs/index.md`)**
- Mission and vision statement
- Architecture workflow diagram (Mermaid)
- Recent decisions table
- Board members with affiliations
- Multiple contact options (Issues, Discussions, Email)
- Citation information

**Assets**
- SciLifeLab symbol as logo (white for header)
- SciLifeLab symbol as favicon (green)

**GitHub Actions (`.github/workflows/deploy-docs.yml`)**
- Automatic deployment to GitHub Pages on push to main/master
- Uses `mkdocs gh-deploy` for publishing

### Preview

Run locally with:
```bash
pip install mkdocs-material
mkdocs serve
```
or check  https://scilifelab.github.io/architecture/

### Checklist

- [x] Implements ADR-0004
- [x] SciLifeLab visual identity applied
- [x] GitHub Actions workflow for deployment
- [x] Cookie consent configured
- [x] CC BY-SA 4.0 license notice in footer
- [x] Contact information (Issues, Discussions, Email)

## GitHub Pages Deployment

The GitHub Action uses `mkdocs gh-deploy` which builds the site and pushes to the `gh-pages` branch.

**Settings:** Keep GitHub Pages configured to "Deploy from branch: `gh-pages`" (no change needed).

**After merging:** Any push to `main` will automatically trigger the workflow, rebuild the site, and update the `gh-pages` branch.

Site URL: https://scilifelab.github.io/architecture/

jhagberg committed:
Add decision document for using MkDocs Material as the documentation site

jhagberg committed:
Add MkDocs configuration, deploy workflow, and initial documentation files

- MkDocs Material theme with SciLifeLab visual identity
- GitHub Actions workflow for automatic deployment
- Homepage with mission, workflow diagram, and board members
- Custom CSS for teal/lime branding
- Cookie consent and Mermaid diagram support

jhagberg committed:
Update README to correct formatting and enhance clarity of technical architecture governance description

jwindhager commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,43 @@
+  push:
+    branches:
+      - main
+      - master
```
```suggestion
```

jwindhager replied:
I don't really care about this specifically, but why was this conversation marked as resolved?

jwindhager commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,43 @@
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
```suggestion
      - uses: actions/checkout@v6
```

jwindhager commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,43 @@
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+
+      - uses: actions/setup-python@v5
```
```suggestion
      - uses: actions/setup-python@v6
```

jwindhager commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,43 @@
+            mkdocs-material-
+
+      - name: Install dependencies
+        run: pip install mkdocs-material
```
perhaps good to use a `requirements.txt` for this, if we later need to add dependencies

jhagberg replied:
yes maybe I just followed this https://squidfunk.github.io/mkdocs-material/publishing-your-site/

jhagberg replied:
I have created a requirments.txt

jwindhager commented on docs/index.md:
````diff
@@ -0,0 +1,83 @@
+    B -->|Guidance| A
+    B -->|Decision| C[ADR]
+    C -->|Published| D[Documentation]
+    D -->|Informs| A
+
+    style A fill:#f5f5f5,stroke:#045C64,stroke-width:2px
+    style B fill:#A7C947,stroke:#8BB030,stroke-width:3px
+    style C fill:#e8f5e9,stroke:#A7C947,stroke-width:2px
+    style D fill:#e8f5e9,stroke:#A7C947,stroke-width:2px
+```
````
Colors make the process a bit hard to read with the current template

jhagberg replied:
The dark template?

jwindhager replied:
yes

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+
+## Recent Decisions
+
+| Decision                                                                                            | Status   | Summary                                    |
+| --------------------------------------------------------------------------------------------------- | -------- | ------------------------------------------ |
+| [ADR-0004: MkDocs Material](decisions/0004-use-mkdocs-material-for-documentation-site.md)           | Accepted | Use MkDocs Material for documentation site |
+| [ADR-0003: Single Process](decisions/0003-use-single-process-approach-for-architecture-requests.md) | Accepted | Single process for architecture requests   |
+| [ADR-0002: CC BY-SA 4.0](decisions/0002-use-cc-by-sa-4-0-license.md)                                | Accepted | Use CC BY-SA 4.0 license for content       |
+| [ADR-0001: Record Decisions](decisions/0001-record-architecture-decisions.md)                       | Accepted | Record architecture decisions as ADRs      |
```
As these are manually listed, I think it would be good to add a checkbox to the ADR PR template

> - [ ] I have listed this ADR in `index.md`

jhagberg replied:
Thanks very good idea! I have added a suggestion for a template!

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+| Johan Viklund     | NBIS        | Technical Expert                                                                                                     |
+| Jonas Söderberg   | NBIS SCoRe  | Technical Expert                                                                                                     |
+| Johannes Alneberg | NGI         | Technical Expert                                                                                                     |
+| Jonas Windhager   | NBIS-BIIF   | Technical Expert                                                                                                     |
```
```suggestion
| Jonas Windhager   | NBIS BIIF   | Technical Expert                                                                                                     |
```

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+| Johannes Alneberg | NGI         | Technical Expert                                                                                                     |
+| Jonas Windhager   | NBIS-BIIF   | Technical Expert                                                                                                     |
+
+_Adjunct members: Björn Nystedt, Hanna Kultima (IDS PMT)_
```
```suggestion
_Adjunct members: Björn Nystedt, Hanna Kultima (Integrated Data Services Project Management Team)_
```

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+
+## Get Involved
+
+The Architecture Board welcomes input from all SciLifeLab teams and the broader community. Whether you have a technical proposal, an architecture concern, or need guidance on a decision, we're here to help.
```
What would be examples for the "broader community" here?

jhagberg replied:
Good catch - that's vague. 

Probably in reality it will be

- 90% will be internal SciLifeLab teams
- 10% external researchers/partners who integrate with our services


Should we make this more specific in the text?

jwindhager replied:
Perhaps "broader academic community in Sweden"?

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+
+Part of [SciLifeLab](https://scilifelab.se) – Sweden's national infrastructure for life sciences research.
+
+**Please cite as:**
+SciLifeLab Architecture Board. (2025). _SciLifeLab Architecture Governance._ GitHub repository: <https://github.com/SciLifeLab/architecture>
```
Personally, I don't need this - but perhaps good for others?

jhagberg replied:
Yes maybe?

libcce replied:
Its just good practice. GitHub is not the best for citation purposes (e.g., versioned releases to Zenodo) but you can evolve this later. For now, something is better than nothing. Also, if you evolve it at some point, there is https://citation-file-format.github.io/. Our policy, at least in the DC, is to also include a license and since this is content CC-BY 4.0 seems appropriate. That guidance should be posted now.

jwindhager replied:
Thanks for chiming in @libcce! I get your point, I'm just not sure anyone would ever realistically want to cite architectural decisions beyond a simple (non-persistent, non-versioned, ...) link :) But I agree, best to lead by example!

jwindhager commented on mkdocs.yml:
```diff
@@ -0,0 +1,102 @@
+site_name: SciLifeLab Architecture Board
```
I think this page should be about the architecture, not about us

```suggestion
site_name: SciLifeLab Architecture
```

jwindhager commented on mkdocs.yml:
```diff
@@ -0,0 +1,102 @@
+nav:
+  - Home: index.md
+  - Architecture Decisions:
+      - decisions/0001-record-architecture-decisions.md
+      - decisions/0002-use-cc-by-sa-4-0-license.md
+      - decisions/0003-use-single-process-approach-for-architecture-requests.md
+      - decisions/0004-use-mkdocs-material-for-documentation-site.md
```
Can we automatically create this? If not, I would suggest adding a checkbox to the ADR PR template:
> - [ ] I have listed this ADR in the `mkdocs.yml` nav section

jhagberg replied:
Yes I have added a suggestion for PR template

jwindhager reviewed (changes requested):
Cool! Some minor suggestions

(only reviewed files specific to ADR-004)

alneberg commented on docs/index.md:
```diff
@@ -0,0 +1,83 @@
+!!! question "Need Architecture Help?"
+
+    - No question too small
+    - Response within 2 weeks
```
I don't think we have any process in place to guarantee this number? What about X-mas and summer vacations?

jhagberg replied:
I have changed it to  "We aim to respond promptly"

alneberg commented:
Looks good! I agree with the comments by @jwindhager .

jhagberg committed:
Apply suggestions from code review

Co-authored-by: Jonas Windhager <jonas@windhager.io>

jhagberg committed:
Update documentation diagrams and styles for improved clarity and branding

- Enhanced Mermaid diagrams with a unified theme for consistency across documentation.
- Updated CSS for primary and secondary button styles to align with the new color scheme.

jhagberg committed:
Add requirements file and update deployment workflow

- Introduced a requirements.txt file to manage dependencies, specifically for mkdocs-material.
- Updated the GitHub Actions workflow to install dependencies from requirements.txt instead of directly installing mkdocs-material.

jhagberg committed:
Update response time language in architecture help section for clarity

jhagberg committed:
Update GitHub link in MkDocs configuration and improve Git credentials formatting in deployment workflow

jhagberg committed:
Update title of decision document to include ADR prefix for consistency

jhagberg committed:
Update documentation to reflect broader academic community engagement in Architecture Board contributions

alneberg commented on README.md:
```diff
@@ -4,6 +4,8 @@
 Technical architecture governance for SciLifeLab infrastructure - ensuring interoperability, standards, and sustainable design
 
+Technical architecture governance for SciLifeLab infrastructure - ensuring interoperability, standards, and sustainable design
+
```
This is just a duplicated line no?

jhagberg replied:
yes

jhagberg committed:
Remove duplicate description in README

Removed duplicate description of technical architecture governance.

jwindhager commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,24 @@
+- [ ] Bug fix
+- [ ] New feature
+- [ ] Documentation update
+- [ ] ADR (Architecture Decision Record)
+
+## ADR Checklist (if applicable)
+
+If this PR includes a new ADR, ensure:
+
+- [ ] ADR is added to `mkdocs.yml` nav section
+- [ ] ADR is added to "Recent Decisions" table in `docs/index.md`
+- [ ] ADR follows the standard template
```
This duplicates the ADR PR template; I suggest removing the redundancy.

Related question: How can I use the ADR PR template to create a PR? I don't think GitHub supports multiple PR templates in their interface, does it?

jhagberg replied:
Yes that  is correct, you have to manually construct a URL to use multiple PR templates. I will delete the adr.md

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,78 @@
+# SciLifeLab Architecture Board
+
+> Ensuring interoperable, sustainable, and FAIR infrastructure across SciLifeLab's federated platforms.
+
+The Architecture Board provides technical governance and guidance for SciLifeLab infrastructure, enabling seamless collaboration while respecting platform autonomy. We support decision-making for technical architecture across all SciLifeLab platforms and services.
```
Personally, I still think that this page should be less focused on us as an entity (that could just be an "About" page), but more on the service/guidance we provide. But not urgent, can discuss/update this later!

jhagberg replied:
Yes that actually sounds good!

jhagberg replied:
Shall we merge this PR first and create a new one where we can refine this or make a new PR to this branch ?

jwindhager commented on docs/index.md:
```diff
@@ -0,0 +1,78 @@
+
+!!! tip "How to Contribute"
+
+    1. **Not sure yet?** Start a [GitHub Discussion](https://github.com/SciLifeLab//architecture/discussions) or email [architecture@scilifelab.se](mailto:architecture@scilifelab.se)
+    2. **Have a question?** Open a [GitHub Issue](https://github.com/SciLifeLab/architecture/issues) to get guidance
+    3. **Proposing a change?** Create an ADR following our [template](decisions/0001-record-architecture-decisions.md)
+    4. **Found an issue?** Submit a pull request with your suggested improvements
```
All of these are world-public. Personally, I think this is a good default, but not everyone may be comfortable with posting publicly, especially when their request is "politically sensitive" (e.g. when their managers already have made up their mind). I'm not sure I have a good suggestion on how to address this barrier while maintaining transparency - perhaps discuss offline at our next meeting?

jhagberg replied:
Yes I guess Email is then the best option.

jhagberg committed:
Remove ADR pull request template

jwindhager reviewed (approved):
I guess we can address the remaining comments separately
